pkg/pillar/cmd/volumemgr/blob.go

```diff
@@ -486,7 +486,12 @@ func populateInitBlobStatus(ctx *volumemgrContext) {
 	}
 	newBlobStatus := make([]*types.BlobStatus, 0)
 	for _, blobInfo := range blobInfoList {
-		mediaType := mediaMap[blobInfo.Digest]
+		mediaType, ok := mediaMap[blobInfo.Digest]
+		if !ok {
+			// if we could not find the media type, we do not know what this blob is, so we ignore it
+			log.Infof("populateInitBlobStatus: blob %s in CAS could not get mediaType", blobInfo.Digest)
+			continue
```

Comment:
So what will happen in this case? Later when we need this particular blob we will download it and hopefully that time we will get a media-type?

Reply:
Correct. Without the media type, the blob is not very useful. And the blob itself never is indicative of the media-type, it always is the layer that called it (manifest or index), and for the root, the Content-Type header, which we store as tags on the image.